### PR TITLE
deprecate metadata on InputContext in favor of definition_metadata

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
This PR deprecates `metadata` in favor of `definition_metadata` and updates all usage in our code. This makes `InputContext` consistent with the deprecation of `metadata` in `OutputContext`

The PR also aligns how definition metadata and user logged metadata are handled in `InputContext` and `OutputContext`. In output context, user logged metadata is kept separate from definition metadata, but in `InputContext` they are merged. I opted to update `InputContext` to keep user logged metadata and definition metadata separate so that `InputContext` and `OutputContext` behave the same way. This also reinforces the distinction between metadata specified at definition time/on the definition object and metadata provided during execution

### CODE REVIEW

**Comment by jamiedemaria** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20206?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#20206** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20206?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#20091** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20091?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#20196](https://github.com/dagster-io/dagster/pull/20196) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20196?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#20198** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20198?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/execution/context/input.py [hunk 1]
```diff
@@ -513,16 +537,28 @@ def test_load_input():
         """
         return self._observations
 
-    def consume_metadata(self) -> Mapping[str, MetadataValue]:
-        result = self._metadata
-        self._metadata = {}
+    def consume_logged_metadata(self) -> Mapping[str, MetadataValue]:
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/execution/context/input.py:540]: renamed this method to align with `OutputContext`. Can revert if we are worried users are calling this method. But it's not part of the public API so i think it should be fine to rename

File: python_modules/dagster/dagster/_core/execution/plan/inputs.py [hunk 2]
```diff
@@ -186,7 +186,10 @@ def load_input_object(
 
         yield from _load_input_with_input_manager(loader, load_input_context)
 
-        metadata = load_input_context.consume_metadata()
+        metadata = {
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/execution/plan/inputs.py:189]: previously, `consume_metadata` returned definition and execution-time metadata as a single dict. It now just returns the execution-time metadata for consistency with `OutputContext` so we need to get the definition metadata as well

File: python_modules/dagster/dagster/_core/execution/context/input.py (hunk 1 above)
Comment (alangenfeld) [reply python_modules/dagster/dagster/_core/execution/context/input.py:540]: is there a reason for users to call this ? seems like its mostly for internal use so i think its fine to change

**Review by alangenfeld — APPROVED** [review]

File: python_modules/dagster/dagster/_core/execution/context/input.py (hunk 1 above)
Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/execution/context/input.py:540]: yeah it's just internal and i dont know of a reason it would be called

**Pull request merged**